### ISSUE TITLE
Emails fail to appear in macOS Outlook redesign

### ISSUE THREAD

**Issue by alexjsteffen** [ProtonMail/proton-bridge#185]
Good day, 

I have been a user of the Protonmail Bridge for some time and I have noticed a persistent issue with the new Microsoft Outlook for macOS [redesign](https://support.microsoft.com/en-us/office/get-help-with-new-outlook-for-mac-64758e05-6cab-4a2d-8c8b-0f3f16ac1cc8?ui=en-US&rs=en-US&ad=US). I have seen this issue on both my old Intel iMac and new M1 MacBook Pro. 

The old design works as expected. 

## Expected Behaviour

- Emails are visible and searchable.
- Send and receive emails.

## Current Behaviour

- Blank screen either saying "loading" or "all emails read". 

**Screenshots:**

<img width="1198" alt="Img_1" src="https://user-images.githubusercontent.com/52205871/112174524-19ece480-8bbc-11eb-9188-7fb6f9f5ca8b.png">
<img width="1172" alt="Img_2" src="https://user-images.githubusercontent.com/52205871/112174547-1ce7d500-8bbc-11eb-8064-fcb3a507b5b3.png">
<img width="767" alt="Img_3" src="https://user-images.githubusercontent.com/52205871/112174554-1f4a2f00-8bbc-11eb-803e-ba7c558fba5a.png">

**Logs:** _This was taken from the log right when I switched Outlook to the redesign and back_

`time="Mar 23 09:02:51.083" level=info msg=FETCH duration=6.879506583s err="<nil>" labelID=5 params="[true 28081:28110 [UID FLAGS INTERNALDATE RFC822.SIZE BODY.PEEK[]]]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:02:56.638" level=info msg=FETCH duration=6.922490291s err="<nil>" labelID=0 params="[true 331:360 [UID FLAGS INTERNALDATE RFC822.SIZE BODY.PEEK[]]]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:02:59.951" level=info msg=FETCH duration=8.45799075s err="<nil>" labelID=5 params="[true 28051:28080 [UID FLAGS INTERNALDATE RFC822.SIZE BODY.PEEK[]]]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:03.672" level=info msg=FETCH duration=6.639258959s err="<nil>" labelID=0 params="[true 301:330 [UID FLAGS INTERNALDATE RFC822.SIZE BODY.PEEK[]]]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:03.851" level=info msg=EXPUNGE duration=164.818042ms err="<nil>" labelID=0 params="[]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:03.851" level=error msg="cannot send response:  write tcp 127.0.0.1:1143->127.0.0.1:52527: write: broken pipe\n" pkg=server-imap
time="Mar 23 09:03:03.851" level=error msg="cannot send response:  write tcp 127.0.0.1:1143->127.0.0.1:52527: write: broken pipe\n" pkg=server-imap
time="Mar 23 09:03:10.095" level=info msg=FETCH duration=9.403310042s err="<nil>" labelID=5 params="[true 28021:28050 [UID FLAGS INTERNALDATE RFC822.SIZE BODY.PEEK[]]]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:10.095" level=error msg="cannot send response:  write tcp 127.0.0.1:1143->127.0.0.1:52524: write: broken pipe\n" pkg=server-imap
time="Mar 23 09:03:10.096" level=error msg="cannot send response:  write tcp 127.0.0.1:1143->127.0.0.1:52524: write: broken pipe\n" pkg=server-imap
time="Mar 23 09:03:10.132" level=info msg=EXPUNGE duration=36.817708ms err="<nil>" labelID=5 params="[]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:10.133" level=error msg="cannot send response:  write tcp 127.0.0.1:1143->127.0.0.1:52524: write: broken pipe\n" pkg=server-imap
time="Mar 23 09:03:10.133" level=error msg="cannot send response:  write tcp 127.0.0.1:1143->127.0.0.1:52524: write: broken pipe\n" pkg=server-imap
time="Mar 23 09:03:11.147" level=info msg=EXPUNGE duration=37.206917ms err="<nil>" labelID=0 params="[]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:11.190" level=info msg=EXPUNGE duration=36.831625ms err="<nil>" labelID=8 params="[]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:11.389" level=info msg=EXPUNGE duration=37.786083ms err="<nil>" labelID=7 params="[]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:17.111" level=info msg=EXPUNGE duration=38.477166ms err="<nil>" labelID=0 params="[]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:22.832" level=info msg=EXPUNGE duration=37.850333ms err="<nil>" labelID=0 params="[]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:28.537" level=info msg=EXPUNGE duration=36.927667ms err="<nil>" labelID=0 params="[]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:28.739" level=info msg=EXPUNGE duration=36.920833ms err="<nil>" labelID=7 params="[]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:28.966" level=info msg=FETCH duration=49.97075ms err="<nil>" labelID=3 params="[true 236:237,419 [UID FLAGS]]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:29.013" level=info msg=EXPUNGE duration=37.461166ms err="<nil>" labelID=3 params="[]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:49.279" level=info msg=FETCH duration=119.966792ms err="<nil>" labelID=5 params="[true 31607:31743,32340:32365,32489,34475:34560 [UID FLAGS INTERNALDATE ENVELOPE BODYSTRUCTURE BODY.PEEK[HEADER.FIELDS (REFERENCES THREAD-TOPIC)]]]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:49.400" level=info msg=FETCH duration=111.767833ms err="<nil>" labelID=5 params="[true 30314:30402,30999:31147,31595:31606 [UID FLAGS INTERNALDATE ENVELOPE BODYSTRUCTURE BODY.PEEK[HEADER.FIELDS (REFERENCES THREAD-TOPIC)]]]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:49.523" level=info msg=FETCH duration=120.735833ms err="<nil>" labelID=5 params="[true 28872:28912,29509:29657,30254:30313 [UID FLAGS INTERNALDATE ENVELOPE BODYSTRUCTURE BODY.PEEK[HEADER.FIELDS (REFERENCES THREAD-TOPIC)]]]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:49.640" level=info msg=FETCH duration=105.116875ms err="<nil>" labelID=5 params="[true 28026:28167,28764:28871 [UID FLAGS INTERNALDATE ENVELOPE BODYSTRUCTURE BODY.PEEK[HEADER.FIELDS (REFERENCES THREAD-TOPIC)]]]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:49.700" level=info msg=EXPUNGE duration=38.232958ms err="<nil>" labelID=5 params="[]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:55.814" level=info msg=FETCH duration=33.474625ms err="<nil>" labelID="o6fSlfuknmZkQfrcQ51epSUYx07uExSidEqZ_A5RSsD_HE1Rx1XoYnKbIecSat3t6n5gp01oFBJrZKFOeBROeA==" params="[true 1815,2357,2549:2550,2789,3007:3008,3011,3210,3212,4003,4015,4313,4320,4327,4343,5000,5004,5349,5515,5537,5736:5738,6073,6075,6251,6615,6780,6782,7148,8205 [UID FLAGS]]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:03:55.902" level=info msg=EXPUNGE duration=36.957583ms err="<nil>" labelID="o6fSlfuknmZkQfrcQ51epSUYx07uExSidEqZ_A5RSsD_HE1Rx1XoYnKbIecSat3t6n5gp01oFBJrZKFOeBROeA==" params="[]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:04:01.598" level=info msg=EXPUNGE duration=36.7855ms err="<nil>" labelID=0 params="[]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="
time="Mar 23 09:04:07.281" level=info msg=EXPUNGE duration=39.284292ms err="<nil>" labelID=0 params="[]" pkg=imap userID="kM_W6w_cKY_Qt6AVfBNjVS5ujVMDAM1DOTf6cDhBtA5iIc6faPkQPS2TMx_YZ-waee2HVMGyvw2MzkIjuvyKHg=="`

## Possible Solution

- N/A

## Steps to Reproduce

1.  Install Outlook for Mac.
2. Setup the bridge per instructions on old design. 
3. Switch to the redesign.  (either on the Beta or other channel that permits seeing "New Outlook") 

_NB:_ I did attempt to sync from the redesign but it either hangs or does the same result as shown above. 

## Version Information

Microsoft Outlook for MacOS (Apple Silicon and Intel): 16.48 (20131707)

## Context (Environment)

- I would like to use the redesign and also access my email. Quite simple. 

Thanks.

**Comment by andrzejsza** [comment]
Yes. We're aware that the new Outlook doesn't support Bridge ;-) Looking into it (for internal reference GODT-1131).

**Comment by scalpol** [comment]
I am having the same exact issue. Sadly I configured a lot of things to be able to use my work outlook account with the new interface so now I can't go back and lost protonmail on my Mac. Any news about it? Has the problem being identified?

**Comment by alexjsteffen** [comment]
I think it may be the way the ProtonMail IMAP implementation is set up because it worked when I proxied and cached the IMAP via [IMAP Proxy](http://www.imapproxy.org). It may make a dent on the email security but the hack worked. Don't know how stable it would be in the long run, however.

**Comment by karlemilnikka** [comment]
Do you have any updates to share regarding this issue? :-)

**Comment by alexjsteffen** [comment]
**Good news!**

It appears to be **working correctly** on the latest insider preview version of Microsoft Office for macOS (M1, Apple Silicon). 

However, in order for it to work correctly, you have to **disable** the use the "Microsoft Cloud" to sync your emails. I am assuming that it's a service similar to Spark email that stores emails on their servers and use their API to sync a lot faster.

TL;DR

- It appears to work with Outlook for macOS version 16.51 (21062701);
- You have to select IMAP and **deselect** the "Microsoft Cloud" sync option;
- Make sure you check the override the default port (if you need to) and select **no** encryption. 

Then you're off to the races!

The small issues I have found that it takes _forever_ to sync a large inbox and it takes awhile to send an email but it _does_ send. I am talking about 10-15 seconds here. 

Cheers,

**Comment by karlemilnikka** [comment]
Thanks for the update @alexjsteffen. Does it still work reliably for you? Version 16.51 is stable now, so I tested with it. Unfortunately, my emails are still missing. I am however running build 21071101 on an Intel Mac.

**Comment by scalpol** [comment]
@karlemilnikka I just tried and it is working (however in a M1 mac).

I had to remove the account and add it again selecting IMAP while disabling the "Microsoft Cloud" option as @alexjsteffen stated. In my case I used STARTTLS and SSL as specified my ProtonMail Bridge, so encryption is working.

Give it a try, good luck!

**Comment by karlemilnikka** [comment]
Thanks for the reply @scalpol. Unfortunately, it doesn’t work for me on an Intel machine. There are still a lot of missing emails (and I’ve tried resetting everything).

**Comment by ledwardchow** [comment]
I'm on an M1 Mac, using STARTTLS. The Inbox and Archive folders sync fine but any that are in named folders/labels do not sync, it will download the first ~5 e-mails, sometimes with bodies, sometimes without.

**Comment by karlemilnikka** [comment]
Another quick update. Microsoft just released version 16.53 of Outlook for Mac. The problem persists. ProtonMail Bridge (1.8.9).

**Comment by karlemilnikka** [comment]
Is this bug acknowledged and will it be addressed? I can confirm that the issue persists in Outlook 16.54 with ProtonMail Bridge 1.8.10.

**Comment by andrzejsza** [comment]
hi @karlemilnikka. the 'bug' here is with Outlook, not Bridge - Bridge still supports IMAP the way it always has.
Outlook for mac is still in active development as a beta product (https://www.microsoft.com/en-us/microsoft-365/roadmap?filters=In%20development%2CMac&searchterms=outlook%2Cfor%2CMac).
before Microsoft calls it a stable product, we would like to avoid spending too much time on this.
but the issues are acknowledged - we know that we need to implement `SEARCH OR` at our end and will keep you posted here once we make progress.

**Comment by karlemilnikka** [comment]
Hi @andrzejsza. I understand. Thanks for the reply.

**Comment by karlemilnikka** [comment]
Do you have any updates to share regarding this issue? The “New Outlook” is now the default Outlook version on macOS.

**Comment by andrzejsza** [comment]
this is considered fixed with V3 - for any other new Outlook related bugs, best to open a new issue.
